doc/11.-Certificate-autoenrollment.md

```diff
@@ -0,0 +1,165 @@
+
+The other settings in this GPO entry do not affect ADSys in any way.
+
+For more advanced configuration, a list of policy servers can be specified in the following GPO entry:
+* Computer Configuration > Policies > Windows Settings > Security Settings > Public Key Policies > Certificate Services Client - Certificate Enrollment Policy
+
+![Certificate advanced configuration](images/Certificate-auto-enrollment/advanced-configuration.png)
+
+## Applying the policy
+
+On the client system, a successful auto-enrollment will place certificate data in the following paths:
+- `/var/lib/adsys/certs` - certificate data
+- `/var/lib/adsys/private/certs` - private key data
+- `/usr/local/share/ca-certificates` - root certificate data (symlinks pointing to `/var/lib/adsys/certs`)
+
+For detailed information on the tracked certificates, `certmonger` can be directly interacted with:
+
+```sh-session
+# Query monitored certificates
+> getcert list
+Number of certificates and requests being tracked: 1.
+Request ID 'galacticcafe-CA.Machine':
+	status: MONITORING
+	stuck: no
+	key pair storage: type=FILE,location='/var/lib/adsys/private/certs/galacticcafe-CA.Machine.key'
+	certificate: type=FILE,location='/var/lib/adsys/certs/galacticcafe-CA.Machine.crt'
+	CA: galacticcafe-CA
+	issuer: CN=galacticcafe-CA,DC=galacticcafe,DC=com
+	subject: CN=keypress.galacticcafe.com
+	issued: 2023-08-18 18:44:27 EEST
+	expires: 2024-08-17 18:44:27 EEST
+	dns: keypress.galacticcafe.com
+	key usage: digitalSignature,keyEncipherment
+	eku: id-kp-clientAuth,id-kp-serverAuth
+	certificate template/profile: Machine
+	profile: Machine
+	pre-save command:
+	post-save command:
+	track: yes
+	auto-renew: yes
+
+# Query known CAs
+> getcert list-cas
+(...)
+CA 'galacticcafe-CA':
+	is-default: no
+	ca-type: EXTERNAL
+	helper-location: /usr/libexec/certmonger/cepces-submit --server=win-mk85nrq26nu.galacticcafe.com --auth=Kerberos
+```
+
+## Policy implementation
+
+With the exception of policy parsing, ADSys leverages the Samba implementation of certificate auto-enrollment. As this feature is only available in newer versions of Samba, we have taken the liberty of vendoring the required Samba files to allow this policy to work on Ubuntu versions that ship an older Samba version. These files are shipped in `/usr/share/adsys/python/vendor_samba`.
+
+To ensure idempotency when applying the policy, we set up a Samba [TDB cache file](https://wiki.samba.org/index.php/TDB) at `/var/lib/adsys/samba/cert_gpo_state_$(hostname).tdb` which contains various information pertaining to the enrolled certificate(s).
+
+Here is an overview of what happens during policy application:
+- GPO parsing (ADSys)
+- execute Python helper script (ADSys)
+- fetch root CA and policy servers (Samba)
+- start monitoring certificate using `certmonger` and `cepces` (Samba)
+
+## Troubleshooting
+
+* Some dependencies are not available in the client Ubuntu installation
```

Comment:
I think it would look better if this was a subtitle (`###`) rather than a list item (`*`) (it looks kind of weird in the formatting).

The same applies to the other ones down below.